safenode/src/domain/storage/disk_backed_record_store.rs

```diff
@@ -227,14 +236,14 @@ impl RecordStore for DiskBackedRecordStore {
         }
     }
 
+    // A backstop replication shall only trigger within pre-defined interval
     fn records(&self) -> Self::RecordsIter<'_> {
-        let mut records = vec![];
-        for key in self.replication_records.iter() {
-            if let Some(record) = self.get(key) {
-                records.push(record);
-            }
+        RecordsIterator {
+            keys: self.records.iter(),
+            storage_dir: self.config.storage_dir.clone(),
+            is_triggered: self.replication_start + self.config.replication_interval
```

Comment:
I don't think we need to return an `RecordsIterator` when we already know here up front weather they will all be returned or not when iterated, isn't it simpler and cheaper to just return an empty iterator if this is `false` here? and otherwise return the `RecordsIterator` ?

Reply:
has to return a `RecordsIterator` holding an `iter` to a non-temp hashset<key>.
will be tricky if not using `keys: self.records.iter()` to calm the clippy erros regarding lifetime.
Hence, prefer to stay with the current.

Reply:
true, then just return `RecordsIterator` with an empty `keys`, that'll be fine.